indico/modules/events/contributions/templates/management/_contribution_list.html

```diff
@@ -108,12 +129,93 @@
                                    data-confirm="{% trans title=contrib.title %}Are you sure you want to completely delete contribution '{{ title }}'?{% endtrans %}"
                                    data-update="#contribution-list"
                                    data-method="DELETE"
-                                   data-href="{{ url_for('.manage_delete_contrib', contrib) }}"></a>
+                                   data-href="{{ url_for('.manage_contrib_rest', contrib) }}"></a>
                             </td>
                         </tr>
                     {% endfor %}
                 </tbody>
             </table>
         </form>
+        <script>
+            (function() {
+                'use strict';
+
+                function patchObject(href, method, data) {
+                    $.ajax({
+                        url: href,
+                        method: method,
+                        data: JSON.stringify(data),
+                        dataType: 'json',
+                        contentType: 'application/json',
+                        error: handleAjaxError,
+                        complete: IndicoUI.Dialogs.Util.progress()
+                    });
+                }
+
+                $('.session-item-picker').itempicker({
+                    containerClasses: 'session-item-container',
+                    filterPlaceholder: $T.gettext('Search for a session'),
+                    createCustomFooterElements: function(container) {
+                        var self = this;
+                        var addNewSession = $('<div>', {
+                            'class': 'action-button',
+                            'text': $T.gettext('Add new session')
+                        });
+
+                        addNewSession.on('click', function() {
+                            self.itempicker('hide');
+                            ajaxDialog({
+                                title: $T.gettext('Add new session'),
+                                url: {{ url_for('sessions.create_session', event) | tojson }},
+                                onClose: function(data) {
+                                    if (data) {
+                                        $('.session-item-picker').itempicker('refreshItemList', data.sessions);
+                                    }
+                                }
+                            });
+                        });
+
+                        container.append(addNewSession);
+                    },
+                    createItem: function(itemData) {
```

Comment:
Well at least the color stuff is a bit special.. not sure how you would implement that nicely without at least some kind of callback. But it would indeed be better if the callback didn't have to create the whole DOM structure for the item.